app/Filament/Admin/Resources/XlsformTemplateResource/RelationManagers/XlsformTemplateLanguageRelationManager.php

```diff
@@ -102,15 +103,51 @@ public function table(Table $table): Table
                         // Get the translation file
                         $uploadedFile = $data['translation_file'];
                         $file = Storage::path($uploadedFile);
-                        
-                        // Create new template language
+                    
+                        // Load the file as an array, getting the first sheet
+                        $rows = Excel::toArray([], $file)[0];
+                    
+                        // Get the headers from the first row and indices for name and new translation columns
+                        $headers = $rows[0];
+                        $nameIndex = array_search('name', $headers);
+                        $newTranslationIndex = array_search('new translation', $headers);
+                    
+                        // Remove the header row
+                        array_shift($rows);
+                    
+                        // Check for missing 'new_translation' when 'name' is not null
+                        $invalidRows = collect($rows)->filter(function ($row) use ($nameIndex, $newTranslationIndex) {
+                            $name = $row[$nameIndex] ?? null;
+                            $newTranslation = $row[$newTranslationIndex] ?? null;
+                            return !empty($name) && (is_null($newTranslation) || trim($newTranslation) === '');
+                        });
+                    
+                        // If there are missing translations, display an error and prevent the import
+                        if ($invalidRows->isNotEmpty()) {
+                            Notification::make()
+                                ->title('Upload unsuccessful')
+                                ->body('The translations file cannot be uploaded as some translations are missing')
+                                ->danger()
+                                ->send();
+                            return;
+                        }
```

Comment:
Logic looks sound here, and it works for me locally. 

The one thing I'd suggest is turning this into custom rule function on the UploadField - that way we can show an inline message when the validation fails, and it won't close the modal (because it won't get to the action() function). See my commit